type-check.ts

```diff
@@ -184,6 +253,16 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<n
       return {a: NONE, tag: stmt.tag, name: stmt.name, value: tValExpr};
     case "expr":
       const tExpr = tcExpr(env, locals, stmt.expr);
+        
+      // code to generate IR for list-comps
+      if(tExpr.tag === "list-comp")
+      {
+        return tcListComp(env,locals,stmt,tExpr);
+      }
+      if(tExpr.tag === "builtin1" && tExpr.arg.tag === "list-comp")
+      {
+        return tcListComp(env,locals,stmt,tExpr.arg);
```

Comment:
What's this case?

Reply:
This case is no longer required and has been updated.